x-pack/plugins/reporting_notifier/public/plugin.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as Rx from 'rxjs';
+import { catchError, filter, map, mergeMap } from 'rxjs/operators';
+import { i18n } from '@kbn/i18n';
+import {
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  PluginInitializerContext,
+} from '../../../../src/core/public';
+import {
+  JOB_COMPLETION_NOTIFICATIONS_POLLER_CONFIG,
+  JOB_COMPLETION_NOTIFICATIONS_SESSION_KEY,
+} from '../constants';
+import { JobId, JobStatusBuckets } from '../index.d';
+import { getGeneralErrorToast } from './components';
+import { ReportingNotifierStreamHandler as StreamHandler } from './lib/stream_handler';
+
+const {
+  jobCompletionNotifier: { interval: JOBS_REFRESH_INTERVAL },
+} = JOB_COMPLETION_NOTIFICATIONS_POLLER_CONFIG;
+
+function getStored(): JobId[] {
+  const sessionValue = sessionStorage.getItem(JOB_COMPLETION_NOTIFICATIONS_SESSION_KEY);
+  return sessionValue ? JSON.parse(sessionValue) : [];
+}
+
+export class ReportingNotifierPublicPlugin implements Plugin<any, any> {
+  private poller$: Rx.Observable<JobStatusBuckets> | null = null;
+
+  // FIXME: License checking: only active, non-expired licenses allowed
+  // Depends on https://github.com/elastic/kibana/pull/44922
+  constructor(initializerContext: PluginInitializerContext) {}
+
+  public setup(core: CoreSetup) {}
+
+  // FIXME: only perform these actions for authenticated routes
+  // Depends on https://github.com/elastic/kibana/pull/39477
+  public start(core: CoreStart) {
+    const { http, notifications } = core;
+    const httpFn = () => http;
+    const notificationsFn = () => notifications;
+    const streamHandler = new StreamHandler(httpFn, notificationsFn);
+
+    this.poller$ = Rx.timer(0, JOBS_REFRESH_INTERVAL).pipe(
```

Comment:
This observable should use a `takeUntil(stop$)` to make sure this gets stopped when the `stop` method is called. See here for an example: https://github.com/elastic/kibana/blob/master/src/core/public/chrome/chrome_service.tsx#L167

Reply:
That is awesome, and it addressed a leftover TODO I still had in this :D

Reply:
89332afbca47d6bcf5f4fdcca05a23218d2607b0